core/src/main/java/org/elasticsearch/index/query/IndicesQueryParser.java

```diff
@@ -87,7 +76,7 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
                         throw new QueryParsingException(parseContext, "[indices] indices or index already specified");
                     }
                     indicesFound = true;
-                    Collection<String> indices = new ArrayList<>();
+                    indices = new ArrayList<>();
```

Comment:
see my comment above, the current error handling is fine but I don't think we shouldn't re-initialize here